### DISCUSSION TITLE
TEKSI developer's guide

### DISCUSSION THREAD

**Discussion by ponceta** [teksi/Home#93] [Q&A]
To be defined :

Actual draft :
https://github.com/teksi/Home/wiki/TEKSI-Developer-Guide

**Comment by ponceta** [comment]
Propositions of today's technical meeting 👍 
QGIS Project
Map templates
QGIS trackable plugin
Custom fields
DB calculated fields
QGIS calculated fields
How to deal with custom symbologies
$variables
naming conventions on views?

Background maps
-> swiss locator maps
WMS / WMTS 

Custom background maps
-> orthophoto / city plans / cadastral data
Vector / WFS / WMS / WMTS

**Comment by rbovard** [comment]
txx_export : Export
txx_sys : System --> Why not `txx_system`? (same for others)

**Reply by sjib** [reply]
see follow-up here: https://github.com/orgs/teksi/discussions/90#discussioncomment-7205842

**Comment by sjib** [comment]
to do: review and integrate: 
https://qgep.github.io/docs/user-guide/layerexplanations/namingconventions.html
https://qgep.github.io/docs/user-guide/layerexplanations/layerexplanation.html
and
https://qgep.github.io/docs/admin-guide/db-customization/index.html

**Reply by sjib** [reply]
see also this discussion: https://github.com/QGEP/datamodel/pull/219#discussion_r1360167974

**Comment by cymed** [comment]
I raise the proposition to distinguish the fields as followed:

1. **DB calculated field:**  fields not present in the interlis model that are used to shortcut topological relations or to prepare labels. (i.e. `tww_od.wastewater_structure._usage_current`). Updated by triggers.  
_**suggested prefix: '\_'**_
2. **Configuration field:**  fields that are not updated by triggers, but that can be used by the db admin to customize the db behaviour. (i.e. `tww_vl.channel_function_hierarchic.order_fct_hierarchic`) 
_**suggested prefix: 'cfg\_'**_
3. **official model extension:**  Official TEKSI model extension to maintain existing functionalities, (i.e. aquifier handling for TEKSI wastewater) 
_**suggested prefix: 'txx\_'**_
4. **Custom field:**  user-specific database extensions not included in the official TEKSI base model, to be excluded from pum. 
_**suggested prefix: 'usr\_'**_

**Reply by cymed** [reply]
Concerning 3: the type of extensions can be added as a comment.
